core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java

```diff
@@ -334,7 +399,72 @@ public FileIO io() {
 
     @Override
     protected String tableName() {
-      return tableIdentifier.toString();
+      return fullTableName;
+    }
+  }
+
+  private class InMemoryViewOperations extends BaseViewOperations {
+    private final FileIO io;
+    private final TableIdentifier identifier;
+    private final String fullViewName;
+
+    InMemoryViewOperations(FileIO io, TableIdentifier identifier) {
+      this.io = io;
+      this.identifier = identifier;
+      this.fullViewName = ViewUtil.fullViewName(catalogName, identifier);
+    }
+
+    @Override
+    public void doRefresh() {
+      String latestLocation = views.get(identifier);
+      if (latestLocation == null) {
+        disableRefresh();
+      } else {
+        refreshFromMetadataLocation(latestLocation);
+      }
+    }
+
+    @Override
+    public synchronized void doCommit(ViewMetadata base, ViewMetadata metadata) {
```

Comment:
It looks like although this is now `synchronized`, there's no protection against concurrently modifying the state fields of the catalog. Any modification to `namespaces`, `tables`, or `views` should be protected.

Reply:
I'm handling this in https://github.com/apache/iceberg/pull/8649. It's currently WIP and requires some additional testing to make sure the inmemory catalog can be safely used in such concurrent conditions

Reply:
I included this now as a separate commit in this PR